inc/WeatherWidget.h

```diff
@@ -0,0 +1,106 @@
+#ifndef WEATHER_WIDGET
+#define WEATHER_WIDGET
+
+#include <string>
+#include "Widget.h"
+#include <raylib.h>
+#include <iostream>
+#include <ctime>
+#include "Constants.h"
+#include "Utility.h"
+#include <json/json.h>
+#include <sstream>
+#include <map>
+using namespace std;
+
+class WeatherWidget : public Widget {
+
+public:
+    /* Docker Checklist
+            Open CMD
+            cd to Smart-Mirror Directory
+            find ip with ipconfig and replace <ip_address> with it
+            run docker run --rm -i -t -v %cd%:/usr/smart-mirror -w /usr/smart-mirror -e DISPLAY=<ip_address>:0.0 smart-mirror
+            you're good to go*/
+    WeatherWidget(uint32_t x, uint32_t y, uint32_t w, uint32_t h, std::string n) :
+        Widget (x, y, w, h, n) { 
+
+        Image weatherIcons = LoadImage("../resources/Adjusted-Weather-Icons.png");
+        ImageResize(&weatherIcons, imageSizeX, imageSizeY);
+        icons = LoadTextureFromImage(weatherIcons);
+        UnloadImage(weatherIcons);
+
+        std::ostringstream api_url;
+        api_url << "https://api.openweathermap.org/data/2.5/weather?lat="
+         << std::to_string(AUSTIN_LATITUDE) << "&lon=" << std::to_string(AUSTIN_LONGITUDE) 
+         << "&appid=" << WEATHER_API;
+        get_http_request(&weatherAPIFinished, api_url.str(), &informations);
+    };
+
+    void draw() override {
+        std::string temp;
+        std::string weather;
+        float ktemp = -1;
+        float ctemp = -1;
+        float ftemp = -1;
+
+        if (weatherAPIFinished) {
+            // Use json here
+            Json::StreamWriterBuilder builder;
+            builder["indentation"] = ""; 
+            // Convert json to string
+            ktemp = stof(Json::writeString(builder, informations["main"]["temp"]));
```

Comment:
On these lines, if you're only doing the API call once, just compute the strings you need to display once. It'll make one frame relatively slow instead of all of them. Space/Time tradeoff, but since it's just strings, just store it in a private class variable